hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/shell/bucket/ListBucketHandler.java

```diff
@@ -50,9 +52,17 @@ protected void execute(OzoneClient client, OzoneAddress address)
     OzoneVolume vol = client.getObjectStore().getVolume(volumeName);
     Iterator<? extends OzoneBucket> bucketIterator =
         vol.listBuckets(listOptions.getPrefix(), listOptions.getStartItem());
-
-    int counter = printAsJsonArray(bucketIterator, listOptions.getLimit());
-
+    List<Object> bucketList = new ArrayList<>();
+    while (bucketIterator.hasNext()) {
```

Comment:
Please keep `limit > counter` from earlier version of the patch.  Limiting the size of the list while it is being populated is better than while printing it.

(Sorry, I was going to make this comment in previous round of review, but somehow forgot to submit.)